fix: display latest effective balance to unauthenticated users

enzo-bitfly opened the pull request:
This adapts to `API` change that sends `latest effective balance` to a public endpoint: `dashboard.validator_dashboards[index].effective_balance` (private) is now `overview.balances.effective_latest` (public)

Also `overview.balances.effective` is now `overview.balances.effective_current`


This PR also fixes [BEDS-1337](https://bitfly1.atlassian.net/browse/BEDS-1337) because the current effective balance is now fetched from the overview, which was already refreshed when adding/removing validators before this change.

[BEDS-1337]: https://bitfly1.atlassian.net/browse/BEDS-1337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

marcel-bitfly commented on frontend/components/dashboard/DashboardValidatorManagementModal.vue:
```diff
@@ -271,43 +271,28 @@ const totalValidators = computed(() => {
-    return effectiveBalanceLimitGuestDashboard
-  }
-  return premium_perks.value?.effective_balance_per_dashboard
+  const freeProduct = products.value?.premium_products.find(product => product.product_name === 'Free')
```
**suggestion (non-blocking):** maybe you expose what you need from `useProductStore()`

enzo-bitfly replied:
Done.
By the way, we don't have type inference for these products but it should be solved with [this API ticket](https://bitfly1.atlassian.net/browse/BEDS-1342)

marcel-bitfly commented on frontend/components/dashboard/DashboardValidatorManagementModal.vue:
```diff
@@ -595,15 +580,15 @@ const inputValidator = ref('')
+                      :value="latestEffectiveBalance ?? '0'"
                       :maximum-fraction-digits="0"
-                      :source-currency="displayCurrencyDefault.main"
+                      :target-currency="displayCurrencyDefault.main"
```
**polish (non-blocking):** you might expose `mainDisplayCurrency` in `BcFormatAmount`

When I have a look at how `BcFormatAmount` is used, it might make sense to adapt and map `mainDisplayCurrency` to `displayCurrencyDefault.main` inside `BcFormatAmount`

enzo-bitfly replied:
Let's discuss this out of the scope of this fix. I don't completely get what you mean.

marcel-bitfly replied:
yeah sure I just meant the same I did with `elDisplayCurrency` for example

enzo-bitfly replied:
I see. Done

marcel-bitfly commented on frontend/pages/dashboard/[[id]]/index.vue:
```diff
@@ -81,7 +81,6 @@ const {
   cookieDashboards,
   dashboards,
 } = storeToRefs(userDashboardStore)
-await useAsyncData('user_dashboards', () => refreshDashboards(), { watch: [ isLoggedIn ] })
```
**question:** this is not needed anymore?

enzo-bitfly replied:
No. It was duplicated, see line 119 of this file.

enzo-bitfly committed:
fix: display latest effective balance to unauthenticated users

- this adapts to API change that sends `latest effective balance` to a public endpoint

See: BEDS-1353
See: BEDS-1337

enzo-bitfly reviewed (commented):
Thank you for the suggestions

marcel-bitfly commented on frontend/stores/useProductsStore.ts:
```diff
@@ -45,6 +45,18 @@ export function useProductsStore() {
+  const premiumProductsByName = computed(() => {
+    if (!data.value?.api_products) return {}
+
+    return Object.fromEntries(data.value.premium_products.map(({
+      product_name,
+      ...rest
+    }) => [
+      product_name,
+      rest,
+    ]))
```
**thought (non-blocking):** yeah you can do this, but there is also nothing wrong with keepin' the `product_name` part of the object (as you do not know how `premiumProductsByName` would be used.

Btw. I would keep the name `premiumProdcuts`, as this is your public api now (but just an opinion -> non-blocking)

enzo-bitfly replied:
I'm renaming to `premiumProducts` and including `product_name` back in

enzo-bitfly committed:
refactor: add prop type to set `displayCurrencyDefault.main` as `target-currency` in `BcFormatAmount.vue`

marcel-bitfly commented on frontend/components/dashboard/DashboardValidatorManagementModal.vue:
```diff
@@ -271,43 +271,28 @@ const totalValidators = computed(() => {
-    return effectiveBalanceLimitGuestDashboard
-  }
-  return premium_perks.value?.effective_balance_per_dashboard
+  const freeProduct = premiumProductsByName.value['Free']
+  const effectiveBalanceLimitFreeProduct = freeProduct?.premium_perks.effective_balance_per_dashboard
```
**suggestion (non-blocking):** you are exposing a `product_name` which might be part of our domain now
```suggestion
  const free = premiumProductsByName.value['Free']
  const effectiveBalanceLimitFree = free?.premium_perks.effective_balance_per_dashboard
```

enzo-bitfly replied:
I prefer to keep the `Product` in, or it would be hard to understand what `free` is if used some lines apart from it's definition

marcel-bitfly reviewed (approved):
nice how with every commit things get cleaner and cleaner 💪
